### PR TITLE
Denoiser Mitsuba-Matrix Fetcher

### PR DESCRIPTION
**Pull request by AnastaZIuk**
status:

- **done**

### CODE REVIEW

File: examples_tests/39.DenoiserTonemapper/main.cpp [hunk 1]
```diff
@@ -73,6 +74,41 @@ int main(int argc, char* argv[])
...
+			auto mesh = meshes_bundle.getContents().begin()[0];
+			auto mesh_raw = static_cast<asset::ICPUMesh*>(mesh.get());
+			const auto mitsubaMetadata = static_cast<const ext::MitsubaLoader::CMitsubaMetadata*>(mesh_raw->getMetadata());
+			const auto data = mitsubaMetadata->getMitsubaMetadata();
+
+			//for(const auto& sensor : data->sensors)
+				//sensor.transform.matrix
+
+			/*
+				Okay, tell me what now please. Is it actually worth to save X matricies for a single scene to .txt?
```
Comment (AnastaZIuk) [inline examples_tests/39.DenoiserTonemapper/main.cpp]: @devshgraphicsprogramming what about those?

File: examples_tests/39.DenoiserTonemapper/exampleInputArguments.txt [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+-COLOR_FILE=../../media/OpenEXR/daily_pt_256.exr -ALBEDO_FILE=../../media/OpenEXR/daily_pt_256.exr -NORMAL_FILE=../../media/OpenEXR/daily_pt_256.exr -MITSUBA_FILE=../../media/mitsuba/daily_pt.xml -MEDIAN_FILTER_RADIUS=1 -DENOISER_EXPOSURE_BIAS=-2.0 -DENOISER_BLEND_FACTOR=0.0 -BLOOM_FOV=0.5 -TONEMAPPER=ACES=0.4,0.8 -OUTPUT=out256_can.exr -COLOR_CHANNEL_NAME=color -ALBEDO_CHANNEL_NAME=albedo -NORMAL_CHANNEL_NAME=normal
+-COLOR_FILE=../../media/OpenEXR/56_render_0_2_16.exr -ALBEDO_FILE=../../media/OpenEXR/56_render_0_2_16.exr -NORMAL_FILE=../../media/OpenEXR/56_render_0_2_16.exr -MITSUBA_FILE=../../media/mitsuba/shapetest.xml -MEDIAN_FILTER_RADIUS=1 -DENOISER_EXPOSURE_BIAS=-2.0 -DENOISER_BLEND_FACTOR=0.0 -BLOOM_FOV=0.5 -TONEMAPPER=ACES=0.4,0.8 -OUTPUT=56_render_0_2_16.exr -COLOR_CHANNEL_NAME=color -ALBEDO_CHANNEL_NAME=albedo -NORMAL_CHANNEL_NAME=normal
```
Comment (AnastaZIuk) [inline examples_tests/39.DenoiserTonemapper/exampleInputArguments.txt]: I'm only not sure whether it is an valid scene we want to fetch a matrix from

File: examples_tests/39.DenoiserTonemapper/CommandLineHandler.hpp [hunk 3]
```diff
@@ -511,7 +494,9 @@ class CommandLineHandler
 		irr::core::vector<std::pair<DENOISER_TONEMAPPER_EXAMPLE_ARGUMENTS,irr::core::vector<float>>> tonemapperBundle;
 		irr::core::vector<std::optional<std::string>> outputFileNameBundle;
 		irr::core::vector<std::optional<std::string>> bloomPsfFileNameBundle;
-		irr::core::vector<std::optional<std::string>> mitsubaFileNameBundle;
+
+		long long elapsedTimeXmls = {};
+		long long elapsedTimeEntireLoading = {};
```
Comment (devshgraphicsprogramming) [inline examples_tests/39.DenoiserTonemapper/CommandLineHandler.hpp]: use chrono types in the future

Comment (AnastaZIuk) [reply examples_tests/39.DenoiserTonemapper/CommandLineHandler.hpp]: fixed

**Comment by devshgraphicsprogramming** [comment]
close this PR, I'm merging it to myself.

**Pull request closed without merging**